pygradle-plugin/src/main/groovy/com/linkedin/gradle/python/util/internal/zipapp/ThinZipappGenerator.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2016 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.gradle.python.util.internal.zipapp;
+
+import com.linkedin.gradle.python.PythonExtension;
+import com.linkedin.gradle.python.util.zipapp.EntryPointTemplateProvider;
+import org.gradle.api.Project;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+
+public class ThinZipappGenerator implements ZipappGenerator {
```

Comment:
Is this class usable directly, or will people have to extend it?

Reply:
They will have to extend it to implement the `buildEntryPoints()` method.  See `ThinPexGenerator`.  This is essentially a refactor of the original `ThinPexGenerator` class for commonality with the internal thin shiv generator class.